__tests__/commands/install/integration.js

```diff
@@ -208,13 +208,17 @@ test.concurrent('hoisting should factor ignored dependencies', async () => {
 });
 
 test.concurrent('--production flag ignores dev dependencies', () => {
-  return runInstall({production: true}, 'install-production', async (config) => {
+  return runInstall({production: true}, 'install-production-without-dev', async (config) => {
```

Comment:
Sounds good. 

The first test was too basic. This new integration test covers a more common case with nested dependencies and dev dependencies.